src/main/scala/fr/polytechnique/cmap/cnam/etl/extractors/ngapacts/NgapActConfig.scala

```diff
@@ -9,7 +9,7 @@ class NgapActConfig(
   def isInCategory(
                     categories: List[NgapActClassConfig],
                     row: Row): Boolean = categories
-      .exists(category => category.naturePrestation == row.getAs[Int]("PRS_NAT_REF") &&
+      .exists(category => category.ngapKeyLetters.contains(row.getAs[String]("PRS_NAT_CB2")) &&
```

Comment:
`row.getAs[String]("PRS_NAT_CB2")`is an expensive operation that you are repeating. You can avoid that by putting the `row.getAs[String]("PRS_NAT_CB2")` outside of the `exists` statement.

Reply:
I tried a solution but I am not sure if this is what you intended.